consensus/src/miner.rs

```diff
@@ -191,7 +193,9 @@ impl MineContext {
             );
         }
 
-        let header = self.find_block(&transactions, &self.canon_header, terminator)?;
+        let ctx = self.clone();
```

Comment:
ditto, why spawn a blocking thread here?

Reply:
general answer provided later on, more specifically the same answer as https://github.com/AleoHQ/snarkOS/pull/1080#discussion_r695011085 applies